proxy.go

```diff
@@ -45,20 +77,36 @@ func (s *Proxy) Proxy(w http.ResponseWriter, r *Request) error {
 		return net.ErrClosed
 	}
 
+	proxyStatus := httpsfv.NewItem(r.Host)
 	addr, err := net.ResolveUDPAddr("udp", r.Target)
 	if err != nil {
-		// TODO(#2): set proxy-status header (might want to use structured headers)
-		w.WriteHeader(http.StatusGatewayTimeout)
+		var dnsError *net.DNSError
+		if errors.As(err, &dnsError) {
+			dnsErrorToProxyStatus(&proxyStatus, dnsError)
+		}
+		proxyStatus.Params.Add("details", err.Error())
+		proxyStatusVal, _ := httpsfv.Marshal(proxyStatus)
+		w.Header().Add("proxy-status", proxyStatusVal)
+
+		w.WriteHeader(errToStatus(err))
 		return err
 	}
+	proxyStatus.Params.Add("next-hop", addr.String())
+
 	conn, err := net.DialUDP("udp", nil, addr)
 	if err != nil {
-		// TODO(#2): set proxy-status header (might want to use structured headers)
-		w.WriteHeader(http.StatusGatewayTimeout)
+		proxyStatus.Params.Add("error", "destination_ip_unroutable")
+		proxyStatus.Params.Add("details", err.Error())
+		proxyStatusVal, _ := httpsfv.Marshal(proxyStatus)
```

Comment:
If you're ignoring the error, please add a comment to explain why this is safe to do. I'm not sure it is though, should we just have a simple error check and send a 5xx on `err != nil`?